subsys/secure_storage/Kconfig.its_store

```diff
@@ -4,12 +4,33 @@
 choice SECURE_STORAGE_ITS_STORE_IMPLEMENTATION
 	prompt "ITS store module implementation"
 
+DT_ITS_PARTITION := $(dt_chosen_path,secure_storage_its_partition)
+DT_CHOSEN_Z_SETTINGS_PARTITION := zephyr,settings-partition
+DT_SETTINGS_PARTITIION := $(dt_chosen_path,$(DT_CHOSEN_Z_SETTINGS_PARTITION))
+DT_STORAGE_PARTITION := $(dt_nodelabel_path,storage_partition)
+
+config SECURE_STORAGE_ITS_STORE_IMPLEMENTATION_ZMS
+	bool "ITS store module implementation using ZMS for storage"
+	depends on FLASH_HAS_DRIVER_ENABLED \
+	 && $(dt_path_enabled,$(DT_ITS_PARTITION)) \
+	 && $(dt_node_has_compat,$(dt_node_parent,$(DT_ITS_PARTITION)),fixed-partitions)
+	select ZMS
+	help
+	  This implementation of the ITS store module makes direct use of ZMS for storage.
+	  It needs a `secure_storage_its_partition` devicetree chosen property that points
+	  to a fixed storage partition that will be dedicated to the ITS. It has lower
+	  overhead compared to the settings-based implementation, both in terms of runtime
+	  execution and storage space, and also ROM footprint if the settings subsystem is disabled.
+	  As this implementations directly maps the PSA storage UIDs to ZMS entry IDs, it limits
+	  their values to the first 30 bits.
+
 config SECURE_STORAGE_ITS_STORE_IMPLEMENTATION_SETTINGS
 	bool "ITS store module implementation using the settings subsystem for storage"
-	DT_STORAGE_PARTITION := $(dt_nodelabel_path,storage_partition)
 	depends on FLASH_HAS_DRIVER_ENABLED \
-		&& $(dt_path_enabled,$(DT_STORAGE_PARTITION)) \
-		&& $(dt_node_has_compat,$(dt_node_parent,$(DT_STORAGE_PARTITION)),fixed-partitions)
+	 && (($(dt_path_enabled,$(DT_SETTINGS_PARTITIION)) \
+	     && $(dt_node_has_compat,$(dt_node_parent,$(DT_SETTINGS_PARTITIION)),fixed-partitions))\
+	  || ($(dt_path_enabled,$(DT_STORAGE_PARTITION)) \
+	      && $(dt_node_has_compat,$(dt_node_parent,$(DT_STORAGE_PARTITION)),fixed-partitions)))
```

Comment:
What would be the correct alignment for you? My logic was to make it so that the `&&` starts at the same column than where the condition it accompanies starts.

Reply:
Your logic is :
depends on A && ((B && C) || (D && E))
My concern of alignment is only on the ||
I would put it
```suggestion
		 && (($(dt_path_enabled,$(DT_SETTINGS_PARTITIION)) \
			 && $(dt_node_has_compat,$(dt_node_parent,$(DT_SETTINGS_PARTITIION)),fixed-partitions))\
			|| ($(dt_path_enabled,$(DT_STORAGE_PARTITION)) \
			    && $(dt_node_has_compat,$(dt_node_parent,$(DT_STORAGE_PARTITION)),fixed-partitions)))
```

Reply:
@tomi-font discard my comment, I tried it locally and there is no perfect way to do this without exceeding the limit of 100 column

Reply:
Yeah I forgot to say, there were actually two constraints to work under here: the 100 columns and the fact that splitting a condition (like `$(dt_node has compat,[...])` into multiple lines doesn't work (I think).